apps/vmq_pulse/src/vmq_pulse_exporter.erl

```diff
@@ -0,0 +1,206 @@
+                E:R ->
+                    error_logger:warning_msg("can't push pulse due to unknown error ~p ~p", [E, R])
+            end;
+        false ->
+            ignore
+    end,
+    erlang:send_after(interval(Interval), self(), push),
+    {noreply, State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% This function is called by a gen_server when it is about to
+%% terminate. It should be the opposite of Module:init/1 and do any
+%% necessary cleaning up. When it returns, the gen_server terminates
+%% with Reason. The return value is ignored.
+%%
+%% @spec terminate(Reason, State) -> void()
+%% @end
+%%--------------------------------------------------------------------
+terminate(_Reason, _State) ->
+    ok.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Convert process state when code is changed
+%%
+%% @spec code_change(OldVsn, State, Extra) -> {ok, NewState}
+%% @end
+%%--------------------------------------------------------------------
+code_change(_OldVsn, State, _Extra) ->
+        {ok, State}.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+create_body(ClusterId, Node) ->
+    Body =
+    [
+        {node, Node},
+        {cluster_id, ClusterId},
+        {pulse_version, ?PULSE_VERSION},
+        {plugins, vmq_plugin:info(all)},
+        {metrics, lists:filtermap(
+                    fun({{metric_def, Type, Labels, _Id, Name, Desc}, Val}) ->
+                            {true, #{type => Type,
+                                     labels => Labels,
+                                     name => Name,
+                                     desc => Desc,
+                                     value => Val}};
+                       (_UnsupportedMetricDef) ->
+                            false
+                    end, vmq_metrics:metrics(#{aggregate => false}))},
+        {applications, [{App, Vsn} || {App, _, Vsn} <- application:which_applications()]},
+        {system_version, erlang:system_info(system_version)},
+        {cluster_status, vmq_cluster:status()},
+        {uname, os:cmd("uname -a")}
+    ],
+    zlib:compress(term_to_binary(Body)).
+
+push_body(PulseUrl, ApiKey, Body) ->
+    ContentType = "application/x-vmq-pulse-1",
+    Headers = case ApiKey of
+                  undefined -> [];
+                  _ ->
+                      AuthStr = base64:encode_to_string(?PULSE_USER ++ ":" ++ ApiKey),
+                      [{"Authorization", "Basic " ++ AuthStr}]
+              end,
+    {ok, ConnectTimeout} = application:get_env(vmq_pulse, connect_timeout),
+    HTTPOpts =  [{timeout, 60000},
+                 {connect_timeout, ConnectTimeout * 1000},
+                 {autoredirect, true}],
+
+    case httpc:request(post, {PulseUrl, Headers, ContentType, Body}, HTTPOpts, []) of
```

Comment:
Can a `httpc` connection be kept open so we don't have to create it every time we post? Reading the `httpc` man page I think this is the default...